x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -23,16 +39,91 @@ export const SAVED_OBJECT_TYPES = {
   VISUALIZATION: 'visualization',
 };
 
+interface RawModuleConfig {
+  id: string;
+  title: string;
+  description: string;
+  type: string;
+  logoFile: string;
+  defaultIndexPattern: string;
+  query: any;
+  jobs: Array<{ file: string; id: string }>;
+  datafeeds: Array<{ file: string; job_id: string; id: string }>;
+  kibana: {
+    search: Array<{ file: string; id: string }>;
+    visualization: Array<{ file: string; id: string }>;
+    dashboard: Array<{ file: string; id: string }>;
+  };
+}
+
+interface MlJobStats {
+  jobs: MlJob[];
+}
+
+interface Config {
+  dirName: any;
+  json: RawModuleConfig;
+}
+
+interface Result {
+  id: any;
+  title: any;
+  query: any;
+  description: any;
+  logo: {
+    icon: string;
+  } | null;
+}
+
+interface JobStat {
+  id: string;
+  earliestTimestampMs: number;
+  latestTimestampMs: number;
+  latestResultsTimestampMs: number;
+}
+
+interface JobExistResult {
+  jobsExist: boolean;
+  jobs: JobStat[];
+}
+
+interface ObjectExistResult {
+  id: string;
+  type: string;
+}
+
+interface ObjectExistResponse {
+  id: string;
+  type: string;
+  exists: boolean;
+  savedObject?: any;
+}
+
+interface SaveResults {
+  jobs: JobResponse[];
+  datafeeds: DatafeedResponse[];
+  savedObjects: KibanaObjectResponse[];
+}
+
 export class DataRecognizer {
-  constructor(callWithRequest) {
-    this.callWithRequest = callWithRequest;
-    this.modulesDir = `${__dirname}/modules`;
-    this.savedObjectsClient = null;
+  modulesDir = `${__dirname}/modules`;
+  indexPatternName: string = '';
+  indexPatternId: string | undefined = undefined;
+  savedObjectsClient: SavedObjectsClient | undefined = undefined;
+
+  callWithRequest: (
+    endpoint: string,
+    clientParams?: Record<string, any>,
+    options?: CallAPIOptions
+  ) => Promise<any>;
+
+  constructor(context: RequestHandlerContext) {
+    this.callWithRequest = context.ml!.mlClient.callAsCurrentUser;
```

Comment:
I think it would be a good idea to name this `this.callAsCurrentUser` to be consistent with the other updated route models